### PR TITLE
Add BSTR xplat PInvoke testcase

### PR DESCRIPTION
**Pull request by luqunl**
Fix for https://github.com/dotnet/coreclr/issues/20182, this PR is about testcase for PInvoke BSTR

Previous these testcases are in https://github.com/dotnet/coreclr/pull/19766, since some teams are blocked by https://github.com/dotnet/coreclr/pull/19766, Thus I checked in product change first in  https://github.com/dotnet/coreclr/pull/19766 and use this PR for testcase

### CODE REVIEW

**Comment by luqunl** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test  please

File: tests/src/Interop/StringMarshalling/BSTR/BSTRTest.cs [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        }
+    }
+    #endregion
+
+    #region ReversePInvoke
+
+    public static string Call_DelMarshal_InOut(string s)
+    {
+        string strRet;
+        if (!s.Equals("ň"))
```
Comment (AaronRobinsonMSFT) [inline tests/src/Interop/StringMarshalling/BSTR/BSTRTest.cs]: @luqunl We should use the unicode value _not_ the actual character. In this case, that would be `\u0148`

**Comment by luqunl** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test please

**Comment by luqunl** [comment]
@dotnet-bot test Windows_NT x64 Checked CoreFX Tests please

File: tests/src/Interop/StringMarshalling/BSTR/BSTRTest.cs [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+using System.Runtime.InteropServices;
+using System;
+using System.Reflection;
+using System.Text;
+using NativeDefs;
+
+class Test
+{
+
+    #region "Reprot Failure"
```
Comment (AaronRobinsonMSFT) [inline tests/src/Interop/StringMarshalling/BSTR/BSTRTest.cs]: Report

**Comment by luqunl** [comment]
@dotnet-bot test this please

**Pull request merged**